awx/main/management/commands/inventory_import.py

```diff
@@ -1032,7 +1064,6 @@ def handle(self, *args, **options):
             else:
                 tb = traceback.format_exc()
                 exc = e
-            transaction.rollback()
```

Comment:
I don't follow this. In the event of a `CommandError`, the exception isn't re-raised, we do `sys.exit(1)` below, so I'd predict that it won't get rolled back if you remove this.

If an inventory update produced a license error, then it would be kind of inconvenient to not roll it back. That could put the user 1,000 hosts over license, and they'd have to go delete a bunch of content somewhere to get out of arrears.

Reply:
A `transaction.rollback()` is illegal when used with `transaction.atomic()`, which is already managing the rollback on exception behavior.  This can be seen if you add a `CommandError` above the line here (https://github.com/ansible/awx/blob/devel/awx/main/management/commands/inventory_import.py#L1005), and then run the `inventory_import` tests.

Reply:
Don't we need to do something else to trigger the rollback then? Either use `set_rollback(True)` or raise (or re-raise) some database exception class?

Reply:
Because we are using `transaction.atomic()` as a context manager, any exception that is raised within that context will trigger the rollback.  The problem, if any, is whether we have bits of code here that need this protection but that are outside of the context manager.

Reply:
> any exception that is raised within that context will trigger the rollback

but we caught the exception (this is something that the documentation says not to do). I don't actually see within this file where this code is wrapped in an atomic block. It's possible that the Django command class is already doing this, that's kind of implied by this code, but it's not a sure-fire bet that the existing code is correct either...

Reply:
I was a little bit fuzzy over whether it was only database exceptions or all exceptions which trigger the rollback, and I feel that Django's docs are a bit ambiguous in this regard, so I went looking:

https://github.com/django/django/blob/7e6b214ed34f5562dbd83cf54924a5b589a29715/tests/transactions/tests.py#L69-L74

It is all exceptions.

Reply:
The relevant protected code is here:

https://github.com/ansible/awx/blob/dd91920c5f858e3335054da3e74ae857724639b0/awx/main/management/commands/inventory_import.py#L1009-L1034

Reply:
ok cool, so if I read correctly, the bottom line here seems to be that it will have already been rolled back by this point, and that line is old junk that didn't do anything?

Reply:
Yep, though worse than that, if it ever did get triggered it would have thrown another error preventing the more graceful bailout